app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.java

```diff
@@ -603,52 +422,57 @@ private void showSearchAndFilterButtons() {
-                    showHideFilter();
-                    if (data.getShowButton()) {
-                        binding.showListBtn.setVisibility(View.VISIBLE);
-                    }
-                }
-            });
+    private void observeScreenState() {
+        viewModel.getScreenState().observe(this, screenState -> {
+            switch (screenState.getScreenState()) {
+                case NONE:
+                    break;
+                case LIST:
+                    configureListScreen((SearchList) screenState);
+                    break;
+                case MAP:
+                    break;
+                case SEARCHING:
+                    configureSearchScreen((SearchForm) screenState);
+                    break;
+                case ANALYTICS:
+                    break;
+            }
+        });
+    }
+
+    private void configureListScreen(SearchList searchConfiguration) {
+        if (switchOpenClose == 1) {
+            showHideFilter();
+        }else if(switchOpenClose == 0){
+            showHideFilterGeneral();
+        }
+
+        syncButtonVisibility(true);
+        setFiltersVisibility(true);
+    }
+
+    private void configureSearchScreen(SearchForm searchConfiguration) {
+        if (switchOpenClose != 1) {
+            showHideFilter();
+        }
+
+        if (searchConfiguration.getQueryHasData()) {
+            binding.clearFilterSearchButton.show();
         } else {
-            liveData.observeForever(searchTeiModels -> {
-                SearchMessageResult data = presenter.getMessage(searchTeiModels);
-                if (data.getMessage().isEmpty()) {
-                    binding.messageContainer.setVisibility(GONE);
-                    binding.scrollView.setVisibility(View.VISIBLE);
-                    liveAdapter.submitList(searchTeiModels);
-                    binding.progressLayout.setVisibility(GONE);
-                } else {
-                    binding.progressLayout.setVisibility(GONE);
-                    binding.messageContainer.setVisibility(View.VISIBLE);
-                    binding.message.setText(data.getMessage());
-                    binding.scrollView.setVisibility(GONE);
-                }
-                CountingIdlingResourceSingleton.INSTANCE.decrement();
-                if (!presenter.getQueryData().isEmpty() && data.getCanRegister())
-                    setFabIcon(false);
-            });
+            binding.clearFilterSearchButton.hide();
         }
-        updateFilters(FilterManager.getInstance().getTotalFilters());
+        syncButtonVisibility(false);
+        setFiltersVisibility(false);
+
+        SearchJavaToComposeKt.setMinAttributesMessage(
+                binding.minAttributeMessage,
+                searchConfiguration
+        );
+    }
+    
+    private void syncButtonVisibility(boolean canBeDisplayed){
+        binding.syncButton.setVisibility(canBeDisplayed ? View.VISIBLE : GONE);
```

Comment:
could this have been binding?

Reply:
Could be. There is more refactor to do in both the activity and the presenter. But I think we should handle those in separate quality tickets.